In [1]:
import pandas as pd
import numpy as np

In [26]:
import chardet
with open(docAddress, 'rb') as f:
    print(chardet.detect(f.read())['encoding'])

UTF-8-SIG


In [32]:
docAddress = r"D:\Data\2019价格监测系统\历史数据\年度产量数据.csv"
df = pd.read_csv(docAddress,encoding='UTF-8-SIG',engine='python')

In [36]:
df.shape

(80571, 36)

In [37]:
df.head()

V1      V2   V3   V4    V5   V6   V7   V8   V9  V10 ...   T10  T11  \
0   大 白 菜  7425.0  0.0  0.0  7425    0  0.0  0.0  0.0  0.0 ...   NaN  NaN   
1   花 椰 菜    24.0  0.0  0.0    24    0  0.0  0.0  0.0  0.0 ...   NaN  NaN   
2  黄    瓜  2853.0  0.0  0.0   843    0  0.0  0.0  0.0  0.0 ...   NaN  NaN   
3  冬    瓜  1376.0  0.0  0.0  1200  NaN  0.0  NaN  NaN  NaN ...   NaN  NaN   
4  茄    子  3580.0  0.0  0.0  1340  NaN  0.0  NaN  NaN  NaN ...   NaN  NaN   

   T12  T13 T14  T15 regionid  quesid  taskdate  row  
0  NaN  NaN NaN  NaN   110112   1-494  20120228    3  
1  NaN  NaN NaN  NaN   110112   1-494  20120228    6  
2  NaN  NaN NaN  NaN   110112   1-494  20120228    9  
3  NaN  NaN NaN  NaN   110112   1-494  20130125   11  
4  NaN  NaN NaN  NaN   110112   1-494  20130125   14  

[5 rows x 36 columns]

In [38]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80571 entries, 0 to 80570
Data columns (total 36 columns):
V1          80571 non-null object
V2          80571 non-null float64
V3          79911 non-null float64
V4          79911 non-null float64
V5          80571 non-null object
V6          62113 non-null object
V7          71140 non-null float64
V8          37233 non-null float64
V9          35797 non-null float64
V10         41805 non-null float64
V11         50234 non-null float64
V12         47200 non-null float64
V13         54181 non-null float64
V14         40590 non-null float64
V15         38913 non-null object
V16         44388 non-null float64
V17         22707 non-null object
T1          0 non-null float64
T2          0 non-null float64
T3          0 non-null float64
T4          0 non-null float64
T5          0 non-null float64
T6          0 non-null float64
T7          0 non-null float64
T8          0 non-null float64
T9          0 non-null float64
T10         0 non-null

In [23]:
df.describe()

V2            V3             V4            V5            V6  \
count  1.729900e+04  1.729900e+04   17299.000000  1.729900e+04  1.540300e+04   
mean   2.301462e+05  2.271427e+05      26.405041  1.764564e+05  1.953467e+05   
std    7.824731e+05  7.746475e+05     959.562955  6.431437e+05  6.711322e+05   
min   -9.464600e+04  0.000000e+00   -2551.970000 -9.564600e+04  0.000000e+00   
25%    6.055150e+03  4.760690e+03      -5.305000  6.500000e+02  2.600000e+03   
50%    4.350000e+04  4.148300e+04       0.000000  2.283708e+04  3.222386e+04   
75%    1.490150e+05  1.465920e+05       5.910000  1.039000e+05  1.187384e+05   
max    1.247709e+07  1.279700e+07  118954.550000  1.083962e+07  1.079607e+07   

                  V7            V8            V9            V10           V11  \
count   16203.000000  1.324900e+04  1.316500e+04   14019.000000  1.508000e+04   
mean       32.621261  1.095857e+04  1.084845e+04      48.161597  3.371699e+04   
std      1235.858397  5.245079e+04  5.224132e+04    1178.112673  1.423631e+05   
min     -2577.880000 -1.788000e+03 -2.000000e+02    -457.140000 -1.650000e+02   
25%        -5.150000  0.000000e+00  0.000000e+00      -4.515000  1.250750e+02   
50%         0.000000  4.450000e+02  4.150000e+02       0.000000  2.230000e+03   
75%         5.490000  4.190050e+03  4.081000e+03       2.830000  1.532000e+04   
max    129660.980000  1.151441e+06  1.151441e+06  117777.500000  3.149027e+06   

           ...        T9  T10  T11  T12  T13  T14  T15       regionid  \
count      ...       0.0  0.0  0.0  0.0  0.0  0.0  0.0   17299.000000   
mean       ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  383900.803515   
std        ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  163927.010035   
min        ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  110000.000000   
25%        ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  230000.000000   
50%        ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  410000.000000   
75%        ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  510000.000000   
max        ...       NaN  NaN  NaN  NaN  NaN  NaN  NaN  650000.000000   

           taskdate           row  
count  1.729900e+04  17299.000000  
mean   2.016487e+07     16.698769  
std    1.575254e+04      8.669079  
min    2.013102e+07      3.000000  
25%    2.015070e+07      9.000000  
50%    2.017010e+07     16.000000  
75%    2.018041e+07     23.000000  
max    2.019070e+07     34.000000  

[8 rows x 33 columns]

### 旬度价格数据
(1430983, 29)

RangeIndex: 1430983 entries, 0 to 1430982
Data columns (total 29 columns):
t1          0 non-null float64
t2          0 non-null float64
t3          0 non-null float64
t4          0 non-null float64
t5          0 non-null float64
t6          0 non-null float64
t7          0 non-null float64
t8          0 non-null float64
t9          0 non-null float64
v1          1430983 non-null object
v2          820002 non-null object
v3          1282075 non-null float64
v4          886601 non-null float64
v5          997784 non-null float64
v6          988513 non-null float64
v7          879555 non-null float64
v8          944812 non-null float64
v9          947952 non-null float64
d1          0 non-null float64
quesid      1430983 non-null object
regionid    1430983 non-null int64
taskdate    1430983 non-null int64
row         1430983 non-null int64
v10         0 non-null float64
v11         0 non-null float64
v12         549644 non-null object
t10         0 non-null float64
t11         0 non-null float64
t12         0 non-null float64
dtypes: float64(22), int64(3), object(4)
memory usage: 316.6+ MB

### 月度数据
(626900, 36)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 626900 entries, 0 to 626899
Data columns (total 36 columns):
V1          626900 non-null object
V2          587471 non-null float64
V3          584528 non-null float64
V4          613056 non-null float64
V5          546088 non-null float64
V6          519299 non-null float64
V7          564640 non-null float64
V8          345403 non-null object
V9          251344 non-null object
V10         254035 non-null float64
V11         50865 non-null object
V12         65711 non-null float64
V13         71868 non-null float64
V14         0 non-null float64
V15         0 non-null float64
V16         0 non-null float64
V17         127759 non-null object
T1          0 non-null float64
T2          0 non-null float64
T3          0 non-null float64
T4          0 non-null float64
T5          0 non-null float64
T6          0 non-null float64
T7          0 non-null float64
T8          0 non-null float64
T9          0 non-null float64
T10         0 non-null float64
T11         0 non-null float64
T12         0 non-null float64
T13         0 non-null float64
T14         0 non-null float64
T15         0 non-null float64
regionid    626900 non-null int64
quesid      626900 non-null object
taskdate    626900 non-null int64
row         626900 non-null int64
dtypes: float64(27), int64(3), object(6)
memory usage: 172.2+ MB

### 季度度数据
(17299, 36)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17299 entries, 0 to 17298
Data columns (total 36 columns):
V1          17299 non-null object
V2          17299 non-null float64
V3          17299 non-null float64
V4          17299 non-null float64
V5          17299 non-null float64
V6          15403 non-null float64
V7          16203 non-null float64
V8          13249 non-null float64
V9          13165 non-null float64
V10         14019 non-null float64
V11         15080 non-null float64
V12         14736 non-null float64
V13         15546 non-null float64
V14         13876 non-null float64
V15         13791 non-null float64
V16         14752 non-null float64
V17         8156 non-null object
T1          0 non-null float64
T2          0 non-null float64
T3          0 non-null float64
T4          0 non-null float64
T5          0 non-null float64
T6          0 non-null float64
T7          0 non-null float64
T8          0 non-null float64
T9          0 non-null float64
T10         0 non-null float64
T11         0 non-null float64
T12         0 non-null float64
T13         0 non-null float64
T14         0 non-null float64
T15         0 non-null float64
regionid    17299 non-null int64
quesid      17299 non-null object
taskdate    17299 non-null int64
row         17299 non-null int64
dtypes: float64(30), int64(3), object(3)
memory usage: 4.8+ MB

### 年度面积数据
(95256, 36)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 95256 entries, 0 to 95255
Data columns (total 36 columns):
V1          95256 non-null object
V2          95256 non-null float64
V3          94596 non-null float64
V4          94596 non-null float64
V5          95256 non-null object
V6          74415 non-null float64
V7          85598 non-null float64
V8          43387 non-null float64
V9          41357 non-null float64
V10         48572 non-null float64
V11         58412 non-null float64
V12         54540 non-null float64
V13         62612 non-null float64
V14         46317 non-null float64
V15         44138 non-null float64
V16         50385 non-null float64
V17         23780 non-null object
T1          0 non-null float64
T2          0 non-null float64
T3          0 non-null float64
T4          0 non-null float64
T5          0 non-null float64
T6          0 non-null float64
T7          0 non-null float64
T8          0 non-null float64
T9          0 non-null float64
T10         0 non-null float64
T11         0 non-null float64
T12         0 non-null float64
T13         0 non-null float64
T14         0 non-null float64
T15         0 non-null float64
regionid    95256 non-null int64
quesid      95256 non-null object
taskdate    95256 non-null int64
row         95256 non-null int64
dtypes: float64(29), int64(3), object(4)
memory usage: 26.2+ MB
### 年度产量数据
(80571, 36)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80571 entries, 0 to 80570
Data columns (total 36 columns):
V1          80571 non-null object
V2          80571 non-null float64
V3          79911 non-null float64
V4          79911 non-null float64
V5          80571 non-null object
V6          62113 non-null object
V7          71140 non-null float64
V8          37233 non-null float64
V9          35797 non-null float64
V10         41805 non-null float64
V11         50234 non-null float64
V12         47200 non-null float64
V13         54181 non-null float64
V14         40590 non-null float64
V15         38913 non-null object
V16         44388 non-null float64
V17         22707 non-null object
T1          0 non-null float64
T2          0 non-null float64
T3          0 non-null float64
T4          0 non-null float64
T5          0 non-null float64
T6          0 non-null float64
T7          0 non-null float64
T8          0 non-null float64
T9          0 non-null float64
T10         0 non-null float64
T11         0 non-null float64
T12         0 non-null float64
T13         0 non-null float64
T14         0 non-null float64
T15         0 non-null float64
regionid    80571 non-null int64
quesid      80571 non-null object
taskdate    80571 non-null int64
row         80571 non-null int64
dtypes: float64(27), int64(3), object(6)
memory usage: 22.1+ MB